test/gradcheck.jl

```diff
@@ -552,7 +552,7 @@ end
 
 @testset "broadcast" begin
   if !Zygote.usetyped
-    @test gradient(x -> sum(sin.(x)), Diagonal(randn(3)))[1][2] == 1
+    @test gradient(x -> sum(sin.(x)), Diagonal(randn(3)))[1][2] == 0
```

Comment:
Is it important? I should probably follow the discussion in https://github.com/FluxML/Zygote.jl/issues/163 but unlike sparse matrices, you can't change the off-diagonal elements in `Digonal`.  So it makes sense to me that off-diagonal gradients are zero.

Reply:
It needs to continue being 1, otherwise it's not consistent with the semantics of other broadcast implementations. See the discussion in #163 for more discussion.